pages/page.py

```diff
@@ -55,55 +62,36 @@ def select_option(self, value, locator):
             raise Exception("Option '" + value + "' was not found, thus not selectable.")
 
     def is_element_present(self, *locator):
-        self.selenium.implicitly_wait(0)
         try:
             self.selenium.find_element(*locator)
             return True
         except NoSuchElementException:
-            # this will return a snapshot, which takes time.
             return False
-        finally:
-            # set back to where you once belonged
-            self.selenium.implicitly_wait(self.testsetup.default_implicit_wait)
 
     def is_element_visible(self, *locator):
```

Comment:
I did this for the wait method because we had two methods: `wait_until_element_visible` and `wait_for_element_visible` that took different arguments. I didn't see a need to modify `is_element_visible` at this time. If you already have an element, you would just need to call `is_displayed`, so I don't think it's really needed there. In my opinion, the wait would best be replaced with a suitable expected condition that could take either an element or locator.